### ISSUE TITLE
Node.js Foundation Technical Steering Committee (TSC) Meeting 2017-10-18

### ISSUE THREAD

**Issue by mhdawson** [nodejs/TSC#384]
## Time

**UTC Wed 18-Oct-2017 17:00 (05:00 PM)**:

| Timezone      | Date/Time             |
|---------------|-----------------------|
| US / Pacific  | Wed 18-Oct-2017 10:00 (10:00 AM) |
| US / Mountain | Wed 18-Oct-2017 11:00 (11:00 AM) |
| US / Central  | Wed 18-Oct-2017 12:00 (12:00 PM) |
| US / Eastern  | Wed 18-Oct-2017 13:00 (01:00 PM) |
| Amsterdam     | Wed 18-Oct-2017 19:00 (07:00 PM) |
| Moscow        | Wed 18-Oct-2017 20:00 (08:00 PM) |
| Chennai       | Wed 18-Oct-2017 22:30 (10:30 PM) |
| Hangzhou      | Thu 19-Oct-2017 01:00 (01:00 AM) |
| Tokyo         | Thu 19-Oct-2017 02:00 (02:00 AM) |
| Sydney        | Thu 19-Oct-2017 04:00 (04:00 AM) |

Or in your local time:
* http://www.timeanddate.com/worldclock/fixedtime.html?msg=Node.js+Foundation+Node.js%20Foundation%20Technical%20Steering%20Committee%20(TSC)+Meeting+2017-10-18&iso=20171018T20
* or http://www.wolframalpha.com/input/?i=8pm+UTC%2C+Oct+18%2C+2017+in+local+time

## Links

* **Minutes Google Doc**: <https://docs.google.com/document/d/1ySnvsuJV3wMnPhaHqBLPv0ESxjUaDdaQarkhmDrD3Mc/edit?usp=sharing>
* Previous minutes: <https://docs.google.com/document/d/1PkoiLPj1b_ePtCL6IWDSwI-eh1Vzp7sfhu5deoiWM-A>_

## Agenda

Extracted from **tsc-agenda** labelled issues and pull requests from the **nodejs org** prior to the meeting.

### nodejs/build

* tmp dir needed on ubuntu 1604 and fedora23 [#873](https://github.com/nodejs/build/issues/873)

### nodejs/node

* Support both OpenSSL 1.1.0 and 1.0.2 [#16130](https://github.com/nodejs/node/pull/16130)
* async_hooks: skip sanity checks when disabled [#15454](https://github.com/nodejs/node/pull/15454)
* buffer: runtime-deprecate Buffer ctor by default [#15346](https://github.com/nodejs/node/pull/15346)
* http: send 400 bad request on parse error [#15324](https://github.com/nodejs/node/pull/15324)
* stream: move prefixed files into internal/streams. [#11957](https://github.com/nodejs/node/pull/11957)
* http2: expose http2 by default, add NODE_NO_HTTP2  [#15684](https://github.com/nodejs/node/pull/15685)

### nodejs/TSC

* Proposal to form a Governance Working Group [#383](https://github.com/nodejs/TSC/issues/383)
* doc: adjust voting rules [#378](https://github.com/nodejs/TSC/pull/378)
* Who should be in @nodejs/security? Who should have access to the private repo? [#358](https://github.com/nodejs/TSC/issues/358)

### nodejs/admin

* doc: doc expectations on TSC and CommComm members [#12](https://github.com/nodejs/admin/pull/12)

## Invited

* Anna Henningsen @addaleax (TSC)
* Ben Noordhuis @bnoordhuis (TSC)
* Сковорода Никита Андреевич @ChALkeR (TSC)
* Colin Ihrig @cjihrig (TSC)
* Evan Lucas @evanlucas (TSC)
* Franziska Hinkelmann @fhinkel (TSC)
* Jeremiah Senkpiel @Fishrock123 (TSC)
* Fedor Indutny @indutny (TSC)
* James M Snell @jasnell (TSC)
* Joyee Cheung @joyeecheung (TSC)
* Matteo Collina @mcollina (TSC)
* Michael Dawson @mhdawson (TSC)
* Brian White @mscdex (TSC)
* Myles Borins @MylesBorins (TSC)
* Ali Ijaz Sheikh @ofrobots (TSC)
* Rod Vagg @rvagg (TSC)
* Shigeki Ohtsu @shigeki (TSC)
* Michaël Zasso @targos (TSC)
* Sakthipriyan Vairamani @thefourtheye (TSC)
* Trevor Norris @trevnorris (TSC)
* Rich Trott @Trott (TSC)
* Josh Gavant @joshgav (TSC)

### Observers
* Bradley Meck @bmeck (GoDaddy/TC39)
* Tracy Hinds @hackygolucky (Node.js Foundation Education Community Manager)
* Kaitlyn Barnard @kbarnard10 (Node.js Foundation Newsletter Curator)
* Mark Hinkle @mrhinkle (Node.js Foundation Executive Director)
* William Kapke @williamkapke (Node.js Community Board representative)

## Notes

The agenda comes from issues labelled with `tsc-agenda` across **all of the repositories in the nodejs org**. Please label any additional issues that should be on the agenda before the meeting starts.

## Joining the meeting

Uberconference; participants should have the link & numbers, contact me if you don't.

## Public participation

We stream our conference call straight to YouTube so anyone can listen to it live, it should start playing at **https://www.youtube.com/c/nodejs+foundation/live** when we turn it on. There's usually a short cat-herding time at the start of the meeting and then occasionally we have some quick private business to attend to before we can start recording & streaming. So be patient and it should show up.

Many of us will be on IRC in #node-dev on Freenode if you'd like to interact, we have a Q/A session scheduled at the end of the meeting if you'd like us to discuss anything in particular. @nodejs/collaborators in particular if there's anything you need from the TSC that's not worth putting on as a separate agenda item, this is a good place f

**Comment by jasnell** [comment]
I would like to invite @dshaw as a regular observer, his input is particularly valuable in governance discussions such as #383, #378 and #358.

**Comment by mhdawson** [comment]
As an FYI these are the tsc-review items for this week

**Comment by fhinkel** [comment]
Are we doing these weekly now instead of bi-weekly?

**Comment by mcollina** [comment]
@fhinkel yes, we decided this some times ago. There seems to be more activity as we are closer to the 9 release.

**Comment by MylesBorins** [comment]
I'd like to invite @sarahnovotny as an observer, they expressed interest at node Interactive and we appeared to have consensus that it was a good idea

**Comment by mcollina** [comment]
👍  for inviting @sarahnovotny

**Comment by mhdawson** [comment]
I'm + 1 for inviting @dshaw and  @sarahnovotny as observers if they are intersted.

**Comment by dshaw** [comment]
Thank you. I am available during the scheduled time and will attend if invited.

**Comment by jasnell** [comment]
@dshaw ... unless there are explicit objections between now and then, please consider yourself invited.

**Comment by Trott** [comment]
https://github.com/nodejs/TSC/pull/378 can be removed from the agenda, I think.

**Comment by Trott** [comment]
> #378 can be removed from the agenda, I think.

Although only if @mhdawson (who put it on the agenda) concurs. Otherwise, it stays on. Those are the rules. :-D

**Comment by Trott** [comment]
> a regular observer

Not objecting but getting this on the radar: I've been considering suggesting that we not have "regular observers" anymore but that observers get invited when it is relevant. I think having regular observers creates a club and implies some elitism TBH, whereas having to specifically think about who to invite each week is probably healthy. It is especially healthy if it causes us to cast our net a little wider on who we invite.

Should we invite bmeck any time we are discussing anything that touches on ES Modules? Absolutely. Should they get invited to every single meeting? I'm not sure that makes sense. (And I suspect Bradley would agree with that assessment.)

**Comment by Trott** [comment]
> Are we doing these weekly now instead of bi-weekly?

@fhinkel Yeah, that was the agreement a month or two ago because we had full agendas for the meetings. The hope was that we'd get back to doing things more asynchronously and being able to cancel most meetings again. (And then we could consider moving to bi-weekly. But we're not there yet, IMO.)

**Comment by Trott** [comment]
Looking at the list of five regularly invited observers, none of them attend regularly. Nor should they, IMO. I'm tempted to add an agenda item saying that we should get rid of regularly invited observers. I'm happy to submit the PR to the tooling that makes the meetings.

**Comment by williamkapke** [comment]
I have always attended regularly. Only recently, with the adoption of the former CTC schedule, has it started to conflict for me.

... but if I've overstayed my welcome- just let me know.

**Comment by Trott** [comment]
@williamkapke I could see making specific role-based exceptions for the two Community Board reps. No one else seems interested in my suggestion yet, so I'm inclined to think we have more urgent things to focus on anyway. It sure seems like I'm speaking for myself and not the TSC as a whole, so keep a-comin' to prove me wrong. :-D

**Comment by mcollina** [comment]
The TSC is chartered to be independent from the Board (if I understand things correctly). That being said, if 1-2 representatives from the Board wants to join our meetings they are definitely welcome.

**Comment by mhdawson** [comment]
With regard to #378 sounds like @MylesBorins still would like it to be on the agenda.

**Comment by mhdawson** [comment]
@Trott I think it would make sense to discuss "regular" observers if you want to add an issue to the agenda.

**Comment by MylesBorins** [comment]
Can we please add https://github.com/nodejs/node/pull/15685 to the agenda

removing flag for HTTP2 on 6.x

**Comment by MylesBorins** [comment]
We should also talk about reverting part of https://github.com/nodejs/node/pull/13322

The zlib optimizations created a number of edges including https://github.com/nodejs/node/issues/14161 and ends up also breaking npm. I realize this is due to userland problems, but unfortunately we find ourselves in a place where we are unable to ship the latest npm on 9.x which is a big issue

**Comment by rvagg** [comment]
Probably won't be attending this one, a bit too early for me.

Removal of "regular observers" makes sense to me too @Trott. We've had a lot of difficulty in the past trying to figure out where to draw the arbitrary line on who gets to attend or not. Particularly when someone tries to invite themselves, it can create a very awkward situation.

That awkwardness is compounded by it being done in public like this too. It's very unlikely that folks are going to want to be totally honest about what they think of such proposals when it involves the very public perception of judging an individual's worth. We've had a lot more candour when we've discussed the merit of adding observers privately than we're going to get by doing it here. Assuming that because nobody objects here to proposals to add specific individuals means that there are no objections is pretty naive IMO. I'm pretty sure you're all aware of the new costs that have been squarely placed on any perception of not being "nice" now; that's going to have a strong chilling effect on honest discussion on topics like this.

We are pretty diligent with streaming meetings these days and trying to be responsive to people outside the meeting that want to participate with Q&A so I'm not seeing a case for including more people on the observers list, you can observe without being dialed in.

I'm -1 on more Foundation representatives (Board or Executive) joining in as participants in these meetings. The toll from administrative work here is already pretty large and I'm hearing a lot of grumbling about how dense the non-technical portions of these meetings are now. Adding more non-technical participants is going to push that in the wrong direction. Plus, technical independence was a founding principle of the Foundation, let's not blur those lines even more than they are already.

**Comment by rvagg** [comment]
@MylesBorins I'm only seeing a single reference to 6.x in the HTTP2 PR https://github.com/nodejs/node/pull/15685, by Sam. So are you making a separate case here for 6.x and shouldn't we perhaps open an issue (maybe in nodejs/release) to talk about that before escalating to TSC?

**Comment by rvagg** [comment]
Re "Invert dependency between core and readable-stream #49":

<img width="721" alt="screen shot 2017-10-18 at 9 37 52 pm" src="https://user-images.githubusercontent.com/495647/31714595-ab71ce8e-b44c-11e7-8f18-ba3b1a4b89f4.png">

My proposal is this: Either a concrete votable proposal is put in front of the TSC by next meeting or it gets taken off the table until one can be formed and someone champions it and brings it back in a more suitable voting form.

**Comment by mcollina** [comment]
> Re "Invert dependency between core and readable-stream #49":

that was waiting for https://github.com/nodejs/TSC/issues/355 to be resolved. When those PRs are landed, I will do a PR against node core. Due to personal matters, this will likely happen between December and January.

In any case, EPS are mothballed so there is no approval needed for that.

**Comment by mcollina** [comment]
@mhdawson I've edited the agenda removing that.

**Comment by jasnell** [comment]
@rvagg... Unlike half of the TSC members, @dshaw and @sarahnovotny both attended the Collaborator Summit and were active participants in the discussions held there that make up a solid half of our agenda today. Their continued input on these discussions makes total sense.

**Comment by williamkapke** [comment]
The observers, @dshaw and @sarahnovotny, were invited to join the discussions at Collab Summit-
 while I was specifically excluded from participating. This cherrypicking/favoritism is unfortunate to see.

Anyhow, it doesn't seem there is consensus on me being an observer any further & my decision to help out with at the board end seems to have confused my interest and participation in the TSC in a most unfortunate way. I decided to sit this one out to give ya'll a chance to discuss without awkwardness from my presence.

**Comment by MylesBorins** [comment]
@williamkapke from my perspective there was no consensus regarding who could observe and who couldn't. Further your name is still included in the list. IMHO you should still join if you are available

**Comment by mhdawson** [comment]
PR for minutes https://github.com/nodejs/TSC/pull/387

**Comment by mhdawson** [comment]
@williamkapke I agree you should still join if you are available.  I'm not sure how the decision was made with respect to the discussion at the Collab Summit, but there has been no agreement to change anything with respect to Observers yet within the TSC and I was not planning on removing you from the invite for upcoming meetings.

**Comment by rvagg** [comment]
Wait, so my objection to additional Foundation observers is just ignored? No attempt at discussion, just write me off because I wasn't at the Collab Summit, no vote?

**Comment by rvagg** [comment]
Opened a new issue for discussion of the topic of sticky regular observers @ #388.

**Comment by rvagg** [comment]
To clarify on specifics here: I have no objection to adding observers if they add value to specific discussion and that appears to be part of the case being made here. I'm objecting to the other part of the case here which is expanding the regular observers list—and since it's difficult to defend the arbitrary adjustment of that list then I think we should just remove it.

**Comment by jasnell** [comment]
Nothing was ignored. @dshaw and @sarahnovotny were directly involved with discussions that were on today's Agenda and therefore had reason to be there. The issue of whether they will be regular observers can be decided separately.

**Comment by rvagg** [comment]
Thanks @jasnell, sorry for the accusatory tone. Will continue this discussion in #388.

**Comment by jasnell** [comment]
> The observers, @dshaw and @sarahnovotny, were invited to join the discussions at Collab Summit- while I was specifically excluded from participating.

To be clear, they were invited to one particular session whose specific purpose was to get specifically their input and a description of the kubernetes governance model in order to inform the ongoing discussions. They were not included in the follow on conversations that were specifically limited to TSC members.

**Comment by mhdawson** [comment]
Meeting help, minutes landed, closing.
